gtsam/gtsam.i

```diff
@@ -2759,21 +2759,35 @@ virtual class EssentialMatrixFactor : gtsam::NoiseModelFactor {
 };
 
 #include <gtsam/slam/dataset.h>
+// Dummy classes, for MATLAB wrappers
+class SfmMeasurement{};
+class SiftIndex{ };
+class SfmMeasurements{};
+class SfmCamera{};
+
 class SfmTrack {
+  SfmTrack();
   Point3 point3() const;
   size_t number_measurements() const;
-  pair<size_t, gtsam::Point2> measurement(size_t idx) const;
-  pair<size_t, size_t> siftIndex(size_t idx) const;
+  void setP(gtsam::Point3& p_);
+  gtsam::SfmMeasurement measurement(size_t idx) const;
+  gtsam::SiftIndex siftIndex(size_t idx) const;
```

Comment:
Do the existing versions of these methods not work? Is there a reason for the change?

Reply:
i propose we come up with a better name than `siftIndex` while we are at it here, since we won't to be using non-SIFT detections, descriptions, and correspondences very soon

Reply:
do we need them at all? Because to my understanding, SiftIndex seems to be the feature coordinates at a particular measurement - then why is SfmMeasurement a pair of <size_t, Point2>? Why not <size_t, SiftIndex>?